src/reaction_lammps_mupt/detectors/reaction_detector.py

```
@@ -296,93 +328,125 @@ def reaction_arranger(monomer_dictionary: dict) -> dict:
-        monomer_dictionary (dict): The same dictionary as input to previous functions.
+        monomer_dictionary: Input dictionary containing monomers + metadata.
 
     Returns:
-        dict: A dictionary containing only the user-selected reactions, keyed by their indices.
+        A dictionary containing only the user-selected reactions, keyed by their indices.
+        If no reactions are detected, raises a ValueError.
+    """
+    arranged_reactions = reaction_arranger(monomer_dictionary)
+    
+    if not arranged_reactions:
+        raise ValueError("There is no possble reactions detected from the given monomers. Terminating program.")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)